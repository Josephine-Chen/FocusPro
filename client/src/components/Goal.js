import React from 'react';
import { Input, Button } from 'react-materialize';
import Nouislider from 'react-nouislider';
import Subgoal from './Subgoal';

// Todo: Replace with better slider
export default class Goal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      subgoals: [],
      subgoal: '',
      step: 1,
    };
    this.handleChange = this.handleChange.bind(this);
    this.postSubgoal = this.postSubgoal.bind(this);
    this.getSubgoals = this.getSubgoals.bind(this);
    this.getStep = this.getStep.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  componentWillMount() {
    this.getSubgoals();
  }

  componentDidMount() {
    this.getSubgoals();
    this.callCustomJQuery();
  }

  getSubgoals() {
    const that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: `/api/goals/${this.props.goal}/subgoals`,
      success: (data) => {
        console.log('SUCCESS: OBTAINED ALL SUBGOALS:', JSON.stringify(data.data));
        that.setState({ subgoals: data.data });
        that.getStep();
      },
      error: (err) => { console.log('ERROR: COULD NOT GET SUBGOALS', err); },
    });
  }

  getStep() {
    if (this.state.subgoals.length > 0) {
      const newStep = Math.round(100 / this.state.subgoals.length);
      this.setState({ step: newStep });
    } else {
      this.setState({ step: 1 });
    }
  }

  handleChange(event) {
    this.setState({ subgoal: event.target.value });
    this.getStep();
  }

  callCustomJQuery() {
    $('.collapsible').collapsible();
  }

  postSubgoal() {
    const that = this;
    $.ajax({
      type: 'POST',
      url: `/api/goals/${this.props.goal}/subgoals`,
      contentType: 'application/json',
      data: JSON.stringify({ subgoal: this.state.subgoal, status: false, GoalId: this.props.goal }),
      success: (data) => {
        console.log('SUCCESS: POSTED INDIVIDUAL GOAL: ', data);
        that.getSubgoals();
        that.cleanInput();
        that.alertConfirmation();
      },
      error: (err) => { console.log('ERROR: COULD NOT POST INDIVIDUAL GOAL', err); },
    });
  }

  alertConfirmation() {
    Materialize.toast('Subgoal set!', 1000);
  }

  cleanInput() {
    this.setState({ subgoal: '' });
  }

  handleKeyPress(e){
    if(e.key == 'Enter'){
      this.postSubgoal();
    }
  }

  render() {
    return (
      <div className="collapsible-body">Progress:
        <br /><br /><br />
        <Nouislider
          range={{ min: 0, max: 100 }}
          start={[0]}
          step={this.state.step}
          behaviour={'tap'}
          format={{
            from: value => `${parseInt(value, 10)}%`,
            to: value => `${parseInt(value, 10)}%`,
          }}
          tooltips
        />
        <br />
        Subgoals:
        <br />
        {this.state.subgoals.length === 0 && <div> There is no subgoal set currently. </div>}
        {this.state.subgoals.map(subgoal =>
          <Subgoal key={`sub ${subgoal.id}`} subgoal={subgoal} status={subgoal.status} id={subgoal.id} user_id={this.props.user_id} goal={subgoal.GoalId} updateSubgoals={this.getSubgoals} />
          )}
        <Input s={8} label="New Subgoal" onChange={this.handleChange} value={this.state.subgoal} onKeyPress={this.handleKeyPress}/>
        <Button className="subgoalButton" waves="light" onClick={this.postSubgoal}>Set Subgoal</Button>
      </div>
    );
  }
}
