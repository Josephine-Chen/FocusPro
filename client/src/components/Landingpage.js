import React from 'react';
import { Row, Col, Card } from 'react-materialize';

// Todo: When using actual images, set max-height and max-width based on % so it scales

export default class Landingpage extends React.Component {
  render() {
    return (
      <div className="row">
        <h3 className="center"> Welcome to FocusPro.</h3>
        <h5 className="center"> We help you limit distractions and stay on task. </h5>
        <br /> <br />
        <Row>
          <Col s={7}>
            <img className="placeholder" src="../../assets/Landingpage-goals.jpeg" alt="goals" />
          </Col>
          <Col s={5}>
            <br /> <br /><br /> <br /> <br />
            FocusPro helps you set goals while tracking your progress in accomplishing them.
            You can blacklist websites to limit distractions by either blocking them directly, or having FocusPro warn you when you have exceeded the time limit you have set.
          </Col>
        </Row>
        <br /> <br />
        <Row>
          <Col s={7}>
            <br /> <br /><br /> <br /> <br />
            Alerts you when you are off task through window notifications using our Chrome extension.
            Shows prompts to ask you to self-reflect on your progress for the day and how much you have worked towards your goal.
          </Col>
          <Col s={5}>
            <img className="placeholder" alt="Chrome extension notification" />
          </Col>
        </Row>
        <br /> <br />
        <Row>
          <Col s={7}>
            <img className="placeholder" src="../../assets/Landingpage-stats.jpeg" alt="stats for site" />
          </Col>
          <Col s={5}>
            <br /> <br /><br /> <br /> <br />
            Intelligently keeps track of overall data patterns.
            See how your habits change as you slowly wean yourself off unproductive sites.
          </Col>
        </Row>
        <br /> <br />
        <Row>
          <Col s={7}>
            <br /> <br /><br /> <br /> <br />
            <h5>Companion Google Chrome Extension</h5>
            Use our Chrome extension to track your tab usage and to send you alert messages and self-reflection questions.
            Download here.
          </Col>
          <Col s={5}>
            <img src="../../assets/icon128.png" alt="Chrome extension icon" />
          </Col>
        </Row>
        <br />
        <div className="center">
          <a className="waves-effect waves-light btn-large">Get Started</a>
        </div>
        <br />
        <h5 className="center"> Meet the Team </h5>
        <Row>
          <Col>
            <Card key={10} className="white" title="Jeff Bentler" actions={[<a href="https://github.com/bentljm">GitHub</a>]}>
              <img src="../../assets/jeff_img.jpeg" className="circle" alt="Jeff" />
            </Card>
          </Col>
          <Col>
            <Card key={11} className="white" title="Josephine Chen" actions={[<a href="https://github.com/Josephine-Chen">GitHub</a>]}>
              <img src="../../assets/josephine_img.jpeg" className="circle" alt="Josephine" />
            </Card>
          </Col>
          <Col>
            <Card key={12} className="white" title="Ian George" actions={[<a href="https://github.com/OppenheimerAndTheMartians">GitHub</a>]}>
              <img src="../../assets/ian_img.jpeg" className="circle" alt="Ian" />
            </Card>
          </Col>
          <Col>
            <Card key={13} className="white" title="Sherry Hsu" actions={[<a href="https://github.com/SherryH">GitHub</a>]}>
              <img src="../../assets/sherry_img.jpeg" className="circle" alt="Sherry" />
            </Card>
          </Col>
        </Row>
      </div>
    );
  }
}
