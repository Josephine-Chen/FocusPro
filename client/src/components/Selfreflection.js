import React from 'react';
import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
import moment from 'moment';
import { getReflectionsAjax } from '../utils/SettingsUtil';

export default class Selfreflection extends React.Component {
  constructor(props) { // Hand downs the prop chain
    super(props);
    this.state = {
      profile: this.props.auth.getProfile(), // it gets imported here
      reflections: [],
      answer: '',
      question: '',
    };
    this.getReflections = this.getReflections.bind(this);
    this.handleCAnswerhange = this.handleAnswerChange.bind(this);

    this.options = {
      defaultSortName: 'date',  // default sort column
      defaultSortOrder: 'asc',  // default sort order
    };
  }

  componentDidMount() {
    this.getReflections();
    this.callCustomJQuery();
  }

  getReflections() {
    const that = this;
    getReflectionsAjax(this.state.profile.user_id, (data) => {
      const reflectionData = data.data.slice();
      // transform data into format accepted by datatable
      (data.data).forEach((refl, index) => {
        reflectionData[index].date = moment(refl.updatedAt).calendar();
      });
      that.setState({
        reflections: reflectionData,
      });
    });
  }

  getCaret(direction) {
    if (direction === 'asc') {
      return ( <i className="fa fa-caret-up" aria-hidden="true" /> );
    }
    if (direction === 'desc') {
      return (
        <i className="fa fa-caret-down" aria-hidden="true" />
      );
    }
    return (
      <i className="fa fa-sort" aria-hidden="true" />
    );
  }

  handleAnswerChange(event) {
    this.setState({
      answer: event.target.value
    });
  }

  callCustomJQuery() {
    $('.collapsible').collapsible();
  }

  render() {
    return (
      <div>
        <h1> Self-Reflection </h1>

        <BootstrapTable data={this.state.reflections} options={this.options} striped hover pagination>
          <TableHeaderColumn hidden isKey dataField="id">ID</TableHeaderColumn>
          <TableHeaderColumn dataSort dataField="question" tdStyle={{ whiteSpace: 'normal' }} caretRender={this.getCaret}>Question </TableHeaderColumn>
          <TableHeaderColumn dataField="answer" tdStyle={{ whiteSpace: 'normal' }}>Answer </TableHeaderColumn>
          <TableHeaderColumn dataSort dataField="date" tdStyle={{ whiteSpace: 'normal' }} caretRender={this.getCaret} width="100">Date </TableHeaderColumn>
        </BootstrapTable>
      </div>
    );
  }
}

