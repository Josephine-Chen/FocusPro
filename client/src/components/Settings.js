import React from 'react';
import { Table, Input, Row, Button, Icon } from 'react-materialize';
import { extend } from 'underscore';

export default class Settings extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      profile: this.props.auth.getProfile(),
      username: '',
      setting: {},
      userId: '',
      blacklist: [],
      siteURL: '',
      siteType: '',
      siteLimit: 0,
      image: '',
      quote: '',
      reminderType: '',
      reminderFreq: 0,
      reminderAddress: '',
      reminderClicked: false,
      labelStyle: {},
      inputStyle: {},
    };
    this.handleReminderSubmission = this.handleReminderSubmission.bind(this);
    this.deleteBlacklist = this.deleteBlacklist.bind(this);
    this.sendNotification = this.sendNotification.bind(this);
    this.siteFormsFilled = this.siteFormsFilled.bind(this);
    this.editStyle = this.editStyle.bind(this);
    this.viewStyle = this.viewStyle.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.handleSubmission = this.handleSubmission.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    this.getUserId();
    this.getSetting();
    this.getBlacklist();
  }

  componentDidUpdate() {
    // this.initialiseSettings();
  }

  getUserId() {
    const that = this;
    $.ajax({
      type: 'GET',
      url: `/api/users/${this.state.profile.user_id}`,
      success: (data) => {
        console.log('SUCCESS: GOT USERID', data.data[0].id);
        that.setState({ userId: data.data[0].id });
        that.setState({ username: data.data[0].username || this.state.profile.nickname });
      },
      error: (err) => { console.log('ERROR: COULD NOT GET USERID', err); },
    });
  }

  getSetting() {
    const that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: `/api/users/${this.state.profile.user_id}/setting`,
      success: (data) => {
        console.log('SUCCESS: OBTAINED SETTINGS: ', data);
        that.setState({ setting: data.data[0] }, () => {
          console.log('get setting', that.state.setting);
          that.initialiseSettings();
        });
        // that.setState({ image: data.data[0].picture });
        // that.setState({ quote: data.data[0].quote });
        that.setState({
          image: data.data[0].picture,
          quote: data.data[0].quote,
          reminderType: data.data[0].reminder_type,
          reminderFreq: data.data[0].reminder_freq,
          reminderAddress: data.data[0].reminder_address,
        });
        // that.initialiseSettings();
      },
      error: (err) => { console.log('ERROR: COULD NOT GET SETTINGS', err); },
    });
  }

  getBlacklist() {
    const that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: `/api/users/${this.state.profile.user_id}/blacklist`,
      success: (data) => {
        console.log('SUCCESS: OBTAINED BLACKLIST: ', data.data);
        that.setState({ blacklist: data.data });
      },
      error: (err) => { console.log('ERROR: COULD NOT GET BLACKLIST', err); },
    });
  }

  updateSetting(pic, quote, refl_freq, remind, remind_type, remind_freq, remind_addr) {
    console.log(remind_freq);
    $.ajax({
      type: 'PUT',
      url: `/api/users/${this.state.profile.user_id}/setting`,
      contentType: 'application/json',
      data: JSON.stringify({
        picture: pic || this.state.setting.picture,
        quote: quote || this.state.setting.quote,
        reflection_freq: refl_freq || this.state.setting.reflection_freq,
        reminder: remind || this.state.setting.reminder,
        reminder_type: remind_type || this.state.setting.reminder_type,
        reminder_freq: remind_freq || this.state.setting.reminder_freq,
        reminder_address: remind_addr || this.state.setting.reminder_address }),
      success: (data) => { console.log('SUCCESS: POSTED SETTING: ', data); },
      error: (err) => { console.log('ERROR: COULD NOT POST SETTING', err); },
    });
  }


  postBlacklist(siteURL, siteType, siteTime) {
    const that = this;
    $.ajax({
      type: 'POST',
      url: `/api/users/${this.state.profile.user_id}/blacklist`,
      contentType: 'application/json',
      data: JSON.stringify({ url: siteURL, blacklist_type: siteType, blacklist_time: siteTime, SettingId: that.state.setting.id }),
      success: (data) => {
        console.log('SUCCESS: POSTED BLACKLIST: ', data);
        that.getBlacklist();
        that.alertUser('Blacklist site');
        that.setState({
          siteURL: '',
          siteType: '',
          siteLimit: 0,
        });
      },
      error: (err) => { console.log('ERROR: COULD NOT GET USERID', err); },
    });
  }

  deleteBlacklist(url_id) {
    const that = this;
    $.ajax({
      type: 'DELETE',
      url: `/api/blacklist/${url_id}`,
      success: (data) => {
        console.log('Sucessfully deleted', data);
        that.getBlacklist();
      },
      error: (err) => { console.log('Error deleting', err); },
    });
  }

  updateUsername(username) {
    const that = this;
    $.ajax({
      type: 'PUT',
      url: `/api/users/${this.state.profile.user_id}/username`,
      contentType: 'application/json',
      data: JSON.stringify({ username }),
      success: () => {
        console.log('SUCCESS: UPDATED USERNAME');
        that.alertUser('Username');
      },
      error: (err) => { console.log('ERROR: COULD NOT GET USERID', err); },
    });
  }

  sendNotification() {
    const that = this;
    $.ajax({
      type: 'POST',
      url: `/api/users/${this.state.profile.user_id}/sendNotification`,
      contentType: 'application/json',
      data: JSON.stringify({ address: this.state.reminderAddress, name: this.state.profile.given_name, freq: this.state.reminderFreq }),
      success: (data) => {
        console.log('SUCCESS: SENT NOTIFICATIONS', data);
        that.alertUser('Email notification');
        that.setState({ reminderClicked: false });
      },
      error: (err) => { console.log('ERROR: COULD NOT SEND NOTIFICATIONS', err); },
    });
  }

  handleReminderSubmission() {
    this.updateSetting(null, null, null, true, this.state.reminderType, this.state.reminderFreq, this.state.reminderAddress);
    this.setState({ reminderClicked: true });
    this.alertUser('Reminder');
  }

  handleChange(event, str) {
    this.setState({ [str]: event.target.value });
  }

  handleSubmission(str) {
    const that = this;
    const delegator = {
      quote: () => {
        that.updateSetting(null, that.state.quote);
        that.alertUser('Quote');
      },
      image: () => {
        that.updateSetting(this.state.image);
        that.alertUser('Image');
      },
      username: () => {
        this.updateUsername(this.state.username.trim());
      },
      site: () => {
        this.postBlacklist(this.state.siteURL, this.state.siteType, this.state.siteLimit);
      },
    };
    delegator[str]();
    that.viewStyle(str);
  }

  handleKeyPress(e, str) {
    // Enter key triggers blur, so don't need to call handleSubmission
    // unless there is no onBlur on the element
    if (e.key === 'Enter') {
      this.viewStyle(str);
      if (str === 'site') {
        if (this.siteFormsFilled()) {
          this.handleSubmission('site');
        }
      }
    }
  }

  siteFormsFilled() {
    return this.state.siteURL.length > 0 && this.state.siteLimit.length > 0 && this.state.siteType.length > 0;
  }

  alertUser(str) {
    Materialize.toast(`${str} set!`, 1000);
  }

  initialiseSettings() {
    const settingList = ['quote'];
    settingList.forEach((item) => {
      if (!this.state.setting[item]) {
        this.editStyle(item);
      } else {
        this.viewStyle(item);
      }
    });

    if (!this.state.setting.picture) {
      this.editStyle('image');
    } else {
      this.viewStyle('image');
    }

    if (!this.state.username) {
      this.editStyle('username');
    } else {
      this.viewStyle('username');
    }
  }

  editStyle(str) {
    this.setState({
      labelStyle: extend(this.state.labelStyle, { [str]: { display: 'none' } }),
      inputStyle: extend(this.state.inputStyle, { [str]: { display: 'block' } }),
    });
  }

  viewStyle(str) {
    this.setState({
      labelStyle: extend(this.state.labelStyle, { [str]: { display: 'block' } }),
      inputStyle: extend(this.state.inputStyle, { [str]: { display: 'none' } }),
    });
  }

  render() {
    const { siteURL, siteLimit, siteType } = this.state;
    let siteSubmitEnabled;
    if (siteURL && siteLimit && siteType) {
      siteSubmitEnabled = siteURL.length > 0 && siteLimit.length > 0 && siteType.length > 0;
    }
    const { reminderType, reminderAddress, reminderFreq } = this.state;
    let reminderSubmitEnabled;
    if (reminderType && reminderAddress && reminderFreq) {
      reminderSubmitEnabled = reminderType.length > 0 && reminderAddress.length > 0 && reminderFreq.length > 0;
    }
    const sendNotificationEnabled = this.state.reminderClicked;

    return (
      <div>
        <h1> Settings </h1>
        <h3> Blacklist: </h3>
        {(this.state.blacklist.length === 0) && <div>There is no blacklist url set currently.</div>}
        {this.state.blacklist.length > 0 && <Table>
          <thead>
            <tr>
              <th data-field="id">Site</th>
              <th data-field="type">Type</th>
              <th data-field="limit">Time Limit</th>
            </tr>
          </thead>
          <tbody>
            {this.state.blacklist.map(site => (
              <tr key={`blacklist${site.id}`} >
                <td>{site.url}</td>
                <td>{site.blacklist_type}</td>
                <td>{site.blacklist_time}</td>
                <td><a href="#/settings" onClick={() => this.deleteBlacklist(site.id)}><Icon right>delete</Icon></a></td>
              </tr>
            ))}
          </tbody>
        </Table>}
        <br />
        <Row>
          <Input s={5} label="Site" value={this.state.siteURL} onChange={e => this.handleChange(e, 'siteURL')} onKeyPress={e => this.handleKeyPress(e, 'site')} />
          <Input s={3} type="select" label="Type" defaultValue="1" value={this.state.siteType} onChange={e => this.handleChange(e, 'siteType')}>
            <option value="1">Blackout</option>
            <option value="2">Block after exceeding</option>
            <option value="3">Warn after exceeding</option>
          </Input>
          <Input s={2} label="Time Limit (min)" value={this.state.siteLimit} onChange={e => this.handleChange(e, 'siteLimit')} onKeyPress={e => this.handleKeyPress(e, 'site')} />
          <Button disabled={!siteSubmitEnabled} className="blacklistButton" waves="light" onClick={() => this.handleSubmission('site')}>Add Site</Button>
        </Row>
        <br />
        <h3> Personalization: </h3>
        <Row>
          <div className="label-header">Username:</div>
          <div onDoubleClick={() => this.editStyle('username')} style={this.state.labelStyle.username}>{this.state.username}
          </div>
          <Input s={10} placeholder="Enter Username" value={this.state.username} onChange={e => this.handleChange(e, 'username')} onKeyPress={e => this.handleKeyPress(e, 'username')} onBlur={() => this.handleSubmission('username')} style={this.state.inputStyle.username} />
        </Row>
        <Row>
          <div className="label-header">Image:</div>
          <div onDoubleClick={() => this.editStyle('image')} style={this.state.labelStyle.image}>{this.state.image}
          </div>
          <Input s={10} placeholder="Enter Image URL" value={this.state.image} onChange={e => this.handleChange(e, 'image')} onBlur={() => this.handleSubmission('image')} onKeyPress={e => this.handleKeyPress(e, 'image')} style={this.state.inputStyle.image} />
        </Row>
        <Row>
          <div className="label-header">Quote:</div>
          <div onDoubleClick={() => this.editStyle('quote')} style={this.state.labelStyle.quote}>{this.state.quote}
          </div>
          <Input s={10} placeholder="Enter Motivational Quote" value={this.state.quote} onChange={e => this.handleChange(e, 'quote')} onBlur={() => this.handleSubmission('quote')} onKeyPress={e => this.handleKeyPress(e, 'quote')} style={this.state.inputStyle.quote} />
        </Row>
        <br />
        <Row>
          <Input s={2} type="select" label="Reminder Type" defaultValue="1" value={this.state.reminderType} onChange={e => this.handleChange(e, 'reminderType')}>
            <option value="1">No Reminder</option>
            <option value="2">Text</option>
            <option value="3">Email</option>
          </Input>
          <Input s={6} label="Number/Email Address" value={this.state.reminderAddress} onChange={e => this.handleChange(e, 'reminderAddress')} />
          <Input s={2} type="select" label="Frequency" defaultValue="1" value={this.state.reminderFreq} onChange={e => this.handleChange(e, 'reminderFreq')}>
            <option value="1">Daily</option>
            <option value="2">Weekly</option>
          </Input>
          <Button disabled={!reminderSubmitEnabled} className="reminderButton" waves="light" onClick={this.handleReminderSubmission}>Set Reminder</Button>
          <Button disabled={!sendNotificationEnabled} waves="light" onClick={this.sendNotification}>Send Notification Now</Button>
        </Row>
        <br />
        <h3> Chrome Extension: </h3>
        Forgot to download the extension? Download it here: ___________
        Your id is {this.state.profile.user_id}. Please enter it into the extension to connect.
      </div>
    );
  }
}
