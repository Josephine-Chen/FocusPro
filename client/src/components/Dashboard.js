import React from 'react';
import {Row, Input, Col, Button, Icon} from 'react-materialize';
import Goal from './Goal.js';
import Site from './Site.js';
import Stat from './Stat.js';
import Motivational from './Motivational.js';

export default class Dashboard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      profile: this.props.auth.getProfile(),
      goals: [],
      goalInput: '',
      dayGoalInput: '',
      userId: '',
      setting: {},
      blacklist: []
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.postGoal = this.postGoal.bind(this);
    this.handleDayGoalChange = this.handleDayGoalChange.bind(this);
    this.handleDayGoalSubmission = this.handleDayGoalSubmission.bind(this);
    this.removeGoal = this.removeGoal.bind(this);
  }

  componentWillMount() {
    this.getUserId();
    this.getSetting();
  }

  componentDidMount(){
    this.getAllGoals();
    this.getBlacklist();
  }

  getUserId() {
    var that = this;
    $.ajax({
      type: 'GET',
      url: 'api/users/' + this.state.profile.user_id,
      success: function (data) {
        console.log("SUCCESS: GOT USER INFO", data.data[0]);
        that.setState({userId: data.data[0].id});
        that.setState({dayGoalInput: data.data[0].daily_goal || ''});
      },
      error: function (err) {
        console.log('ERROR: COULD NOT GET USERID', err);
      }
    });
  }

  getSetting() {
    var that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: '/api/users/' + this.state.profile.user_id + '/setting',
      success: function(data) {
        console.log("SUCCESS: OBTAINED SETTINGS: ", data);
        that.setState({setting: data.data});
      },
      error: function(err) {console.log("ERROR: COULD NOT GET SETTINGS", err);}
    });
  }

  getBlacklist() {
    var that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: '/api/users/' + this.state.profile.user_id + '/setting/blacklist',
      success: function(data) {
        console.log("SUCCESS: OBTAINED BLACKLIST: ", data.data);
        that.setState({blacklist: data.data});
      },
      error: function(err) {console.log("ERROR: COULD NOT GET BLACKLIST", err);}
    });
  }

  getAllGoals () {
    var that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: '/api/users/' + this.state.profile.user_id + '/goals/',
      success: function(data) {
        console.log("SUCCESS: OBTAINED ALL GOALS:", data);
        that.setState({goals: data.data});
      },
      error: function(err) {
        console.log("ERROR: COULD NOT GET ALL GOALS", err);
      }
    });
  }

  handleInputChange(event) {
    this.setState({goalInput: event.target.value});
  }

  handleDayGoalChange(event) {
    this.setState({dayGoalInput: event.target.value});
  }

  handleDayGoalSubmission() {
    $.ajax({
      type: 'PUT',
      url: '/api/users/' + this.state.profile.user_id,
      contentType: 'application/json',
      data: JSON.stringify({daily_goal: this.state.dayGoalInput}),
      success: function(data) {console.log("Update daily goal to", data);},
      error: function(err) {console.log("Error updating daily goal", err);}
    });
  }

  postGoal() {
    var that = this;
    $.ajax({
      type: 'POST',
      url: '/api/users/' + this.state.profile.user_id + '/goals',
      contentType: 'application/json',
      data: JSON.stringify({goal: this.state.goalInput, progress: 0, goal_picture: '', UserId: this.state.userId}),
      success: function(data) {
        console.log("SUCCESS: POSTED INDIVIDUAL GOAL: ", data);
        that.setState({goalInput: ''}); //Not clearing input...
        that.getAllGoals();
      },
      error: function(err) {console.log("ERROR: COULD NOT POST INDIVIDUAL GOAL", err);}
    });

  }

  removeGoal (goal_id) {
    var that = this;
    $.ajax({
      type: 'DELETE',
      url: '/api/users/' + this.state.profile.user_id + '/goals/' + goal_id,
      success: function(data) {
        console.log("Remove goal:", data);
        that.getAllGoals();
      },
      error: function(err) {
        console.log("ERROR: COULD NOT GET ALL GOALS", err);
      }
    });
  }

  render() {
    return (
      <div>
        <h1> Welcome, {this.state.profile.given_name} </h1>
        {this.state.setting.quote}
        <br />
        <br />
        <h3> Goal of the Day: </h3>
        <Row>
          <Input s={10} value={this.state.dayGoalInput} onChange={this.handleDayGoalChange} /> <Button className="dayGoalButton" waves='light' onClick={this.handleDayGoalSubmission}>Save</Button>
        </Row>
        <h3> Main Goals: </h3>
          <ul className="collapsible" data-collapsible="expandable">
          {this.state.goals.map((goal, index) => (
            <li key = {index}>
            <div className="collapsible-header">{goal.goal} <a href = '#/dashboard' onClick = {()=>this.removeGoal(goal.id)}><Icon right>delete</Icon></a></div>
            <Goal key = {'goal' + index} goal = {goal.id} user_id = {this.state.profile.user_id} />
          </li>
            ))}
        </ul>
        <Row>
        <Input s={8} label="New Goal" value={this.state.value} onChange={this.handleInputChange} /> <Button className="goalButton" waves='light' onClick={this.postGoal}>Set Goal</Button> <Motivational />
        </Row>
        <br />
        <h3> Sites: </h3>
        <ul className="collapsible" data-collapsible="expandable">
          {this.state.blacklist.map((site, index) => (
            <li key = {'sites ' + index}>
              <div className="collapsible-header">{site.url}</div>
              <div className="collapsible-body"><Site url = {site.url} siteId = {site.id}/></div>
            </li>
            ))}
        </ul>
        <br />
        <br />
        <h3> Stats: </h3>
        <Row>
          <Col s={4}>
          <Stat />
          Today's Stats
          </Col>
          <Col s={4}>
          <Stat />
          Week's Stats
          </Col>
          <Col s={4}>
          <Stat />
          All Time Stats
          </Col>
        </Row>
      </div>
    );
  }
}