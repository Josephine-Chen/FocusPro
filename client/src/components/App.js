import React, { Component } from 'react';
import Landingpage from './Landingpage.js';
import {Sidebar} from './Sidebar.js';
import Auth from './Auth.js';

export default class App extends React.Component {

  constructor (props) {
    super(props);
  }

  componentWillMount () {
      this.lock = new Auth0Lock('V3oJOmHRIeW7nJnLZIa7ioWOo8godJVQ', 'focuspro.auth0.com');
      // Set the state with a property that has the token
      this.setState({idToken: this.getIdToken()})
  }

  createLock () {
    this.lock = new Auth0Lock(this.props.clientId, this.props.domain);
  }

  getIdToken () {
    // First, check if there is already a JWT in local storage
    var idToken = localStorage.getItem('id_token');
    var authHash = this.lock.parseHash(window.location.hash);
    // If there is no JWT in local storage and there is one in the URL hash,
    // save it in local storage
    if (!idToken && authHash) {
      if (authHash.id_token) {
        idToken = authHash.id_token
        localStorage.setItem('id_token', authHash.id_token);
      }
      if (authHash.error) {
        // Handle any error conditions
        console.log("Error signing in", authHash);
      }
    }
    return idToken;
  }

  render() {
    return (
      <div>
        <Sidebar />
        <main>
          <div className="container">
            <Auth lock={this.lock} />
            {this.props.children}
          </div>
        </main>
      </div>
    );
  }
}



