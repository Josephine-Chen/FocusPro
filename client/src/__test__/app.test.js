import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App.js';
import Landingpage from '../components/Landingpage.js';
import Dashboard from '../components/Dashboard.js';
import Goal from '../components/Goal.js'
import Subgoal from '../components/Subgoal.js'
import Selfreflection from '../components/Selfreflection.js';
import Settings from '../components/Settings.js';

describe('App (Snapshot)', () => {
  it('App renders', () => {
    // const component = renderer.create(<App />);
    // const json = component.toJSON();
    // expect(json).toMatchSnapshot();
  });
});

describe('Landingpage (Snapshot)', () => {
  it('Landingpage renders', () => {
    const component = renderer.create(<Landingpage />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Dashboard (Snapshot)', () => {
  it('Dashboard renders', () => {
    const component = renderer.create(<Dashboard />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Goal (Snapshot)', () => {
  it('Goal renders', () => {
    const component = renderer.create(<Goal />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Subgoal (Snapshot)', () => {
  it('Subgoal renders', () => {
    const component = renderer.create(<Subgoal />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Selfreflection (Snapshot)', () => {
  it('Selfreflection renders', () => {
    const component = renderer.create(<Selfreflection />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Settings (Snapshot)', () => {
  it('Settings renders', () => {
    const component = renderer.create(<Settings />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});


