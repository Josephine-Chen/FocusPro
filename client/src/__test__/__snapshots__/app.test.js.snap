exports[`App (Snapshot) App renders 1`] = `<div />`;

exports[`Dashboard (Snapshot) Dashboard renders 1`] = `
<div>
  <h1>
     Welcome, USER 
  </h1>
  <div>
    <h3>
       Goal 
    </h3>
    <div>
      <h5>
         Subgoal 
      </h5>
    </div>
  </div>
</div>
`;

exports[`Goal (Snapshot) Goal renders 1`] = `
<div>
  <h3>
     Goal 
  </h3>
  <div>
    <h5>
       Subgoal 
    </h5>
  </div>
</div>
`;

exports[`Landingpage (Snapshot) Landingpage renders 1`] = `
<div>
  <h1>
     FocusPro 
  </h1>
  <h5>
     Welcome to PhocusPro. 
    <br />
     We help you limit distractions and stay on task. 
  </h5>
</div>
`;

exports[`Selfreflection (Snapshot) Selfreflection renders 1`] = `
<div>
  <h1>
     Self-Reflection 
  </h1>
</div>
`;

exports[`Settings (Snapshot) Settings renders 1`] = `
<div>
  <h1>
     Settings 
  </h1>
</div>
`;

exports[`Subgoal (Snapshot) Subgoal renders 1`] = `
<div>
  <h5>
     Subgoal 
  </h5>
</div>
`;
